Services/Diagnostics/DiagnosticsLogger.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Http;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Runtime;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics
+{
+
+    public interface IDiagnosticsLogger
+    {
+        Task<IHttpResponse> LogServiceStartAsync();
+
+        Task<IHttpResponse> LogServiceHeartbeatAsync();
+
+        Task<IHttpResponse> LogServiceErrorAsync(
+            string message,
+            [CallerMemberName] string callerName = "",
+            [CallerFilePath] string filePath = "",
+            [CallerLineNumber] int lineNumber = 0);
+
+        Task<IHttpResponse> LogServiceErrorAsync(
+            string message,
+            Exception e,
+            [CallerMemberName] string callerName = "",
+            [CallerFilePath] string filePath = "",
+            [CallerLineNumber] int lineNumber = 0);
+
+        Task<IHttpResponse> LogServiceErrorAsync(
+            string message,
+            object data,
+            [CallerMemberName] string callerName = "",
+            [CallerFilePath] string filePath = "",
+            [CallerLineNumber] int lineNumber = 0);
+    }
+
+    public class DiagnosticsLogger : IDiagnosticsLogger
+    {
+        public struct JsonStruct
+        {
+            public string EventId;
+            public string EventType;
+            public string DeploymentId;
+            public string SolutionType;
+            public Dictionary<string, object> EventProperties;
+
+            public JsonStruct(string st1, string st2, string st3, string st4, Dictionary<string, object> eventProps)
```

Comment:
remove 1,2,3,4 from variable names. Give them more meaningful names

Reply:
I see for myself that I am naming variables in a random manner. Will take extra care from next time. Sorry about this.